### PR TITLE
Dissassemble Event Object Data

### PR DESCRIPTION
**Pull request by garakmon**

### CODE REVIEW

File: data/event_object_movement.s [hunk 1]
```diff
@@ -5352,485 +6432,3137 @@ gMapObjectGraphicsInfo_83A5134:: @ 83A5134
 	.4byte gSpriteImages_83A1CF0
 	.4byte gDummySpriteAffineAnimTable
 
-gUnknown_83A5158:: @ 83A5158
-	.incbin "baserom.gba", 0x3A5158, 0xB0
+gObjectEventSpritePalettes:: @ 83A5158
+	.4byte gUnknown_836D828
+	.2byte 0x1103
```
Comment (PikalaxALT) [inline data/event_object_movement.s]: `obj_pal gUnknown_836D828, 4355`

File: data/event_object_movement.s [hunk 2]
```diff
@@ -5352,485 +6432,3137 @@ gMapObjectGraphicsInfo_83A5134:: @ 83A5134
...
+	obj_image_anim_jump 0
+
+gUnknown_83A5B70:: @ 83A5B70
+	.4byte gUnknown_83A5B54
+
+gUnknown_83A5B74:: @ 83A5B74
+	spr_template 0xFFFF, 0x1004, gEventObjectBaseOam_16x16, gUnknown_83A5B70, gUnknown_83A5B34, gDummySpriteAffineAnimTable, sub_80DCD1C
+
+gUnknown_83A5B8C:: @ 83A5B8C
+	obj_rot_scal_anim_frame 0xFF00, 256, 128, 0
```
Comment (PikalaxALT) [inline data/event_object_movement.s]: use signed decimals for the x and y args to `obj_rot_scal_anim_frame`

**Comment by garakmon** [comment]
I'm marking as ready. I'll change to decimal numbers when I decompile the data in the next week or so.

**Comment by jiangzhengwenjz** [comment]
There're still lots of output with `egrep -inr \(mapobject\|fieldobject\) src`

**Comment by garakmon** [comment]
okay should be fixed now

**Comment by jiangzhengwenjz** [comment]
`include/overworld.h:35:extern struct LinkPlayerObjectEvent gLinkPlayerMapObjects[4];`

File: asm/shop.s [hunk 3]
```diff
@@ -0,0 +1,3563 @@
+	.include "asm/macros.inc"
```
Comment (jiangzhengwenjz) [inline asm/shop.s]: This file is no longer in use

**Comment by garakmon** [comment]
graphics/map_objects doesn't exist for me

**Comment by jiangzhengwenjz** [comment]
My bad. More issue I found: 
- `MAP_OBJ_GFX_` also needs to be changed
-  some header guard macros need to be changed, e.g. `GUARD_MAP_OBJ_80688E4_H`, `GUARD_MAP_OBJ_8097404_H`, `GUARD_MAP_OBJ_LOCK_H`,

File: src/fldeff_rocksmash.c [hunk 4]
```diff
@@ -106,7 +106,7 @@ static void sub_80C99A0(u8 taskId)
 
 bool8 SetUpFieldMove_RockSmash(void)
 {
-    if (CheckObjectGraphicsInFrontOfPlayer(/*MAP_OBJ_GFX_BREAKABLE_ROCK*/0x60) == TRUE)
+    if (CheckObjectGraphicsInFrontOfPlayer(/*OBJECT_EVENT_GFX_BREAKABLE_ROCK*/0x60) == TRUE)
```
Comment (jiangzhengwenjz) [inline src/fldeff_rocksmash.c]: `OBJECT_EVENT_GFX_ROCK_SMASH_ROCK`

File: src/fldeff_strength.c [hunk 5]
```diff
@@ -7,14 +7,14 @@
...
-#include "constants/map_objects.h"
+#include "constants/object_events.h"
 
 static void FldEff_UseStrength(void);
 static void sub_80D08A8(void);
 
 bool8 SetUpFieldMove_Strength(void)
 {
-    if (TestPlayerAvatarFlags(PLAYER_AVATAR_FLAG_SURFING) || CheckObjectGraphicsInFrontOfPlayer(/*MAP_OBJ_GFX_PUSHABLE_BOULDER*/0x61) != TRUE)
+    if (TestPlayerAvatarFlags(PLAYER_AVATAR_FLAG_SURFING) || CheckObjectGraphicsInFrontOfPlayer(/*OBJECT_EVENT_GFX_PUSHABLE_BOULDER*/0x61) != TRUE)
```
Comment (jiangzhengwenjz) [inline src/fldeff_strength.c]: `OBJECT_EVENT_GFX_STRENGTH_BOULDER`

**Pull request merged**